lib/internal/modules/esm/get_format.js

```diff
@@ -92,6 +92,12 @@ let typelessPackageJsonFilesWarnedAbout;
 function getFileProtocolModuleFormat(url, context = { __proto__: null }, ignoreErrors) {
   const { source } = context;
   const ext = extname(url);
+  const deduceFormat = (fromSource, fromUrl) => {
+    const { existsSync, readFileSync } = require('fs');
+    const realSource = fromSource ?? existsSync(fromUrl) ? readFileSync(fromUrl).toString() : undefined;
```

Comment:
Using `existSync` is an anti-pattern (the file could get deleted between the moment when `existsSync` returned true and the call to `readFileSync`, i.e. a race condition), instead we should just call `readFileSync` and catch the error – and we should probably not call `.toString` unless there's a good reason to.
```suggestion
    let realSource = fromSource;
    try {
      realSource ??= readFileSync(fromUrl);
    } catch {
      return 'commonjs';
    }
```